applications/FluidDynamicsApplication/python_scripts/navier_stokes_solver_vmsmonolithic.py

```diff
@@ -311,21 +356,36 @@ def Initialize(self):
 
         KratosMultiphysics.Logger.PrintInfo("NavierStokesSolverMonolithic", "Solver initialization finished.")
 
+    def InitializeSolutionStep(self):
+        if self._TimeBufferIsInitialized():
+            # If required, compute the BDF coefficients
+            if hasattr(self, 'bdf_process'):
+                (self.bdf_process).Execute()
+            # Perform the solver InitializeSolutionStep
+            (self.solver).InitializeSolutionStep()
 
     def _set_physical_properties(self):
-        # Transfer density and (kinematic) viscostity to the nodes
-        for el in self.main_model_part.Elements:
-            rho = el.Properties.GetValue(KratosMultiphysics.DENSITY)
-            if rho <= 0.0:
-                raise Exception("DENSITY set to {0} in Properties {1}, positive number expected.".format(rho,el.Properties.Id))
-            dyn_viscosity = el.Properties.GetValue(KratosMultiphysics.DYNAMIC_VISCOSITY)
-            if dyn_viscosity <= 0.0:
-                raise Exception("DYNAMIC_VISCOSITY set to {0} in Properties {1}, positive number expected.".format(dyn_viscosity,el.Properties.Id))
-            kin_viscosity = dyn_viscosity / rho
-            break
-
-        KratosMultiphysics.VariableUtils().SetScalarVar(KratosMultiphysics.DENSITY, rho, self.main_model_part.Nodes)
-        KratosMultiphysics.VariableUtils().SetScalarVar(KratosMultiphysics.VISCOSITY, kin_viscosity, self.main_model_part.Nodes)
+        # Check if fluid properties are provided using a .json file
+        materials_filename = self.settings["material_import_settings"]["materials_filename"].GetString()
+        if (materials_filename != "unknown_materials.json"):
+            # Add constitutive laws and material properties from .json file to model parts
+            material_settings = KratosMultiphysics.Parameters("""{"Parameters": {"materials_filename": ""}} """)
+            material_settings["Parameters"]["materials_filename"].SetString(materials_filename)
+            KratosMultiphysics.ReadMaterialsUtility(material_settings, self.model)
+        else:
+            # Get density and dynamic viscostity from the properties and transfer them to the nodes
+            for el in self.main_model_part.Elements:
```

Comment:
>     * Why main-model-part and not computing-model-part?
Fair question, this was there before these changes so I probably make this mistake when programming the first version of the solver. We can take the change to correct it.
>     * Why only one element? What happens of some elements have different properties?
In fluid dynamics problems to have different properties means to have different materials, which turns to be a two-fluids problems. We always have a unique property.
>     * Why not accessing directly the Properties of the Modelpart?
If I'm not wrong, to access the properties from the ModelPart you need the properties of interest identifier, which is not known by the solver nor the .json settings file. This is why we the decided to retrieve the properties of interest from the elements.

Reply:
I've checked why we use the `main_model_part` instead of the `computing_model_part`. When the physical properties are set, the `computing_model_part` is not created yet.

Reply:
ok thx for explaining

btw I think when you specify through json you also have to give the property-ID => then it would be known
Not sure though

Reply:
You're right, when you provide it using json you specify the property Id., but we want to keep retrocompatibility with the old cases (which didn't use the json).